http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpCromwellDAO.scala

```diff
@@ -0,0 +1,96 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import akka.http.scaladsl.model.StatusCode
+import cats.effect.Async
+import cats.mtl.Ask
+import cats.syntax.all._
+import io.circe._
+import org.broadinstitute.dsde.workbench.leonardo.dao.HostStatus.HostReady
+import org.broadinstitute.dsde.workbench.leonardo.dns.RuntimeDnsCache
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.leonardo.{CloudContext, RuntimeName}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.http4s._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.client.middleware.{Retry, RetryPolicy}
+import org.typelevel.log4cats.Logger
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+class HttpCromwellDAO[F[_]](val runtimeDnsCache: RuntimeDnsCache[F],
+                            httpClient: Client[F],
+                            samDAO: SamDAO[F]
+)(implicit logger: Logger[F], F: Async[F], metrics: OpenTelemetryMetrics[F])
+    extends CromwellDao[F]
+    with Http4sClientDsl[F] {
+
+  private val retryable = (_: Request[F], resp: Either[Throwable, Response[F]]) => resp match {
+    case Left(_) => true
+    case Right(r: Response[F]) => ???
+  }
+  private val retryPolicy = RetryPolicy[F](RetryPolicy.exponentialBackoff(30 seconds, 5), retryable)
+  private val httpClientWithRetry = Retry(retryPolicy)(httpClient)
+
+  import HttpCromwellDAO._
+
+  override def getStatus(cloudContext: CloudContext, runtimeName: RuntimeName)(implicit
+    ev: Ask[F, TraceId]
+  ): F[CromwellStatusCheckResponse] =
+    for {
+      hostStatus <- Proxy.getRuntimeTargetHost[F](runtimeDnsCache, cloudContext, runtimeName)
+      headers <- cloudContext match {
+        case _: CloudContext.Azure =>
+          samDAO.getLeoAuthToken.map(x => Headers(x))
+        case _: CloudContext.Gcp =>
+          F.pure(Headers.empty)
+      }
+      res <- hostStatus match {
+        case host: HostReady =>
+
+          metrics.incrementCounter("cromwell/status") >>
+            httpClientWithRetry.expectOr[CromwellStatusCheckResponse](
+            Request[F](
+              method = Method.GET,
+              uri = host.toUri.withPath(Uri.Path.unsafeFromString(s"api/engine/v1/status")),
```

Comment:
This calls Cromwell, but I think we'll want to aggregate status for WDS and CBAS status endpoints as well. The Leo "app" should only come online if all 3 services are OK status.

Not sure if we want to do that aggregation in `HttpCromwellDAO`, or make `HttpWdsDAO` and `HttpCbasDAO` classes as well…?

Reply:
Yeah I am thinking we do the aggregation in AKSInterpreter